### ISSUE TITLE
i18n-extract generates incompatible XLIFF

### ISSUE THREAD

**Issue by fetis** [angular/angular#40235]
# 🐞 bug report

### Affected Package
@angular/localize

### Is this a regression?
Yes, it worked fine in v10

### Description
After upgrade to v11, XLIF export is not accepted by our translation tool anymore. We use the Lokalise platform for translation and it reports missed placeholders in files from v11, the same codebase from v10 produces zero errors.

Here's an example of an error message

![Screen Shot 2020-12-22 at 16 08 13](https://user-images.githubusercontent.com/577042/102903522-be9e3f80-4470-11eb-9964-5a7fc42c56b8.png)

Why do I think it's an Angular problem? I see that the generated output has changed from version 10 to 11, so this might be the reason for issue. I'm not sure is it a valid output from the XLIFF perspective and I would like to clarify it from both sides (Angular and Lokalise)

## 🔬 Minimal Reproduction
here's the source code for translation
```
    <p i18n>Please note,<br>
            that a phone call is required to fix your created case as soon as possible.<br>
            Please call our repair service at <a href="tel:{{repairNumber.international}}">{{repairNumber.humanReadable}}</a>,<br>
            and provide your case number {{caseId}}.</p>
```

here's the XLIFF output from Angular 10

```
<trans-unit id="cffc0f93e757e51e8c783232713584e9c65d35f6" datatype="html">
  <source>Please note,<x id="LINE_BREAK" ctype="lb" equiv-text="&lt;br/&gt;"/>
      that a phone call is required to fix your created case as soon as possible.<x id="LINE_BREAK" ctype="lb" equiv-text="&lt;br/&gt;"/>
      Please call our repair service at <x id="START_LINK" ctype="x-a" equiv-text="&lt;a&gt;"/><x id="INTERPOLATION" equiv-text="{{repairNumber.humanReadable}}"/><x id="CLOSE_LINK" ctype="x-a" equiv-text="&lt;/a&gt;"/>,<x id="LINE_BREAK" ctype="lb" equiv-text="&lt;br/&gt;"/>
      and provide your case number <x id="INTERPOLATION_1" equiv-text="{{caseId}}"/>.</source>
  <context-group purpose="location">
...
  </context-group>
</trans-unit>
```

and here's the same output from Angular 11
```
<trans-unit id="cffc0f93e757e51e8c783232713584e9c65d35f6" datatype="html">
  <source>Please note,<x id="LINE_BREAK" ctype="lb" equiv-text="&lt;br&gt;"/> that a phone call is required to fix your created case as soon as possible.<x id="LINE_BREAK" ctype="lb" equiv-text="&lt;br&gt;"/> Please call our repair service at <x id="START_LINK" ctype="x-a" equiv-text="&lt;a href=&quot;tel:{{repairNumber.international}}&quot;&gt;"/>
  <x id="INTERPOLATION" equiv-text="{{repairNumber.humanReadable}}"/><x id="CLOSE_LINK" ctype="x-a" equiv-text="&lt;/a&gt;"/>,
  <x id="LINE_BREAK" ctype="lb" equiv-text="&lt;br&gt;"/> and provide your case number <x id="INTERPOLATION_1" equiv-text="{{caseId}}"/>.
</source>
  <context-group purpose="location">
...
  </context-group>
</trans-unit>
```

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 11.0.5
Node: 12.18.3
OS: darwin x64

Angular: 11.0.5
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1100.5
@angular-devkit/build-angular   0.1100.5
@angular-devkit/core            11.0.5
@angular-devkit/schematics      11.0.5
@angular/cdk                    11.0.3
@angular/material               11.0.3
@schematics/angular             11.0.5
@schematics/update              0.1100.5
rxjs                            6.6.3
typescript                      4.0.3
</code></pre>

**Comment by petebacondarwin** [comment]
Hi @fetis - the difference that appears to be causing the problem is that this placeholder:

```xml
<x id="START_LINK" ctype="x-a" equiv-text="&lt;a&gt;"/>
```

is changed to:

```xml
<x id="START_LINK" ctype="x-a" equiv-text="&lt;a href=&quot;tel:{{repairNumber.international}}&quot;&gt;"/>
```

specifically the `href` is now included in the `equiv-text` property.

This is completely valid XLIFF, AFAICT. So I believe that it is actually a bug in the processing of the Lokalise platform.

...

I created an evaluation account and I saw that it is complaining about this, as your issue describes. But note that the tool appears to be unescaping the `<a` part of the `equiv-text` attribute value and treating it as a new element rather than part of the attribute.

<img width="612" alt="Screenshot 2020-12-22 at 15 47 56" src="https://user-images.githubusercontent.com/15655/102906932-39fdf200-446d-11eb-858a-a41df74deeff.png">

...

So this is working as expected and I suggest opening an issue with the Lokalise tool.
